### PR TITLE
[brgrbic] SqlOD changes for cloud Server Type and minor bug fixes

### PR DESCRIPTION
**Pull request by BranislavGrbicMDCS**
…minimal set of changes needed in sqltoolservice in order to keep everything working within ADS.
Removing access to Server roles.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs [hunk 1]
```diff
@@ -25,9 +26,9 @@ public class ListDatabasesRequestDatabaseProperties
     /// </summary>
     static class ListDatabaseRequestHandlerFactory
     {
-        public static IListDatabaseRequestHandler getHandler(bool includeDetails, bool isSqlDB)
+        public static IListDatabaseRequestHandler getHandler(bool includeDetails, bool isSqlDB, ConnectionInfo connectionInfo = null)
         {
-            if (!includeDetails)
+            if (!includeDetails || connectionInfo?.EngineEdition == DatabaseEngineEdition.SqlOnDemand)
```
Comment (BranislavGrbicMDCS) [inline src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs:31]: Comment from Aleksandar's PR (https://github.com/microsoft/sqltoolsservice/pull/1090/files) is resolved here.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeGenerator.cs [hunk 2]
```diff
@@ -265,7 +265,7 @@ protected override void OnExpandPopulateFolders(IList<TreeNode> currentChildren,
                 NodeType = "Folder",
                 NodeTypeId = NodeTypes.ServerLevelServerRoles,
                 IsSystemObject = false,
-                ValidFor = ValidForFlag.AllOnPrem|ValidForFlag.SqlOnDemand,
+                ValidFor = ValidForFlag.AllOnPrem,
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeGenerator.cs:268]: Is the SqlOnDemand flag being used elsewhere? If not remove it

**Comment by Charles-Gagnon** [comment]
No SMO update needed for any of this?

**Comment by BranislavGrbicMDCS** [comment]
> No SMO update needed for any of this?

There is SMO change that Aleksandar checked in. He has reached out to David regarding SMO change. So yes, this and ADS change is depending on those SMO changes.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeGenerator.cs (hunk 2 above)
Comment (BranislavGrbicMDCS) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeGenerator.cs:268]: Yup, it is used on several places in the same file.

**Review by Charles-Gagnon — APPROVED** [review]
Signing off with the assumption that you've tested this locally and verified things are working as expected.

**Comment by BranislavGrbicMDCS** [comment]
> Signing off with the assumption that you've tested this locally and verified things are working as expected.

Yes, everything tested locally, working as expected.

**Pull request merged**